org-code-javabuilder/protocol/src/main/java/org/code/protocol/ClientMessage.java

```diff
@@ -67,4 +67,12 @@ public String getFormattedMessage() {
     }
     return formattedMessage.toString();
   }
+
+  /**
+   * @return whether or not this message should always be sent by any output adapter. Some messages
+   *     are only relevant for the test output adapter
+   */
+  public boolean shouldAlwaysSend() {
```

Comment:
I know this is already merged (and no issues with it being merged as is because I know it was fixing a prod issue) but had a couple thoughts if we want to consider a follow up -

I think the naming of `shouldAlwaysSend` feels a little confusing because the logic specifically pertains to the UserTestOutputAdapter. If we want to stick with this approach, maybe naming it something like `sendInTestsOnly()` or something might be clearer?

Another thought that could work here, especially if we might have other cases where we skip specific output adapters: 
We could add a `AdapterType/String getType()` to the `OutputAdapter` interface, and then have a method on `ClientMessage` like `List<String/AdapterType> adaptersToSkip()` that returns which adapter types the message should not be sent for. That way, if WebSocket and AWS both return the same type, something like `MAIN` or `USER`, vs if UserTest returns `TEST`, the Painter messages could signal that they want to skip the `MAIN/USER` adapters but send to all others (and by default no adapters are skipped). The logic could also be reversed where we explicitly list which adapter types are allowed (and by default all adapter types are allowed).

Let me know what you think - happy to file a JIRA as a follow-up if either of these make sense.